File: 13.py
class ListNode(object):
	"""docstring for ListNode"""
	def __init__(self, value):
		#super(ListNode, self).__init__()
		self.next = None
		self.value = value
def delete_node(phead,pdelete):
	if not phead or not pdelete:
		return False

	if pdelete.next:
		pdelete.value = pdelete.next.value
		pdelete.next = pdelete.next.next
	elif phead == pdelete:
		phead = None
	else:
		pdelete = None

node1 = ListNode(10)
node2 = ListNode(11)
node3 = ListNode(13)
node4 = ListNode(15)
node1.next = node2
node2.next = node3
node3.next = node4
delete_node(node1, node3)
print(node3.value)
delete_node(node1, node1)
print(node1.value)

File: 13.cpp
#include<iostream>
using namespace std;
struct ListNode{
	int value;
	ListNode* next;
};
void deleteNode(ListNode* head,ListNode* node){
	//若没有元素

	//若删除尾部
	if(node->next==NULL){
	    ListNode* pNode=head;
		while(pNode->next!=node){
			pNode=pNode->next;		
		}
		pNode->next = NULL;
		delete node;
	}
	else if(head == node){
		head->next = NULL;
		delete node;
	}
	else{

	    //有多个元素
	    node->value = node->next->value;
		node->next = node->next->next;
	}
}

int main(){
	ListNode* a =  new ListNode;
	a->value=3;
	ListNode* b =  new ListNode;
	ListNode* c =  new ListNode;
	b->value=4;
	c->value=5;
	a->next = b;
	b->next = c;
	c->next = NULL;
	deleteNode(a,b);
	cout<<a->next->value;
	int stop;
	cin>>stop;
	return 1;
}